core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -174,7 +174,8 @@ class LogCleaner(initialConfig: CleanerConfig,
   }
 
   override def validateReconfiguration(newConfig: KafkaConfig): Unit = {
-    val numThreads = LogCleaner.cleanerConfig(newConfig).numThreads
+    val newCleanerConfig = LogCleaner.cleanerConfig(newConfig)
+    val numThreads = newCleanerConfig.numThreads
```

Comment:
Is there a reason why we changed this?

Reply:
It is not intended to keep this change, which was made while debugging a test.